packages/onboarding/src/setup-tailored-site-after-creation.ts

```diff
@@ -37,74 +35,61 @@ interface SetupOnboardingSiteOptions {
 
 export function setupSiteAfterCreation( { siteId, flowName }: SetupOnboardingSiteOptions ) {
 	// const { resetOnboardStore } = dispatch( ONBOARD_STORE );
-	const { saveSiteSettings, setIntentOnSite, setStaticHomepageOnSite } = dispatch( SITE_STORE );
 	const selectedPatternContent = select( ONBOARD_STORE ).getPatternContent();
 	const siteTitle = select( ONBOARD_STORE ).getSelectedSiteTitle();
 	const siteDescription = select( ONBOARD_STORE ).getSelectedSiteDescription();
 	const siteLogo = select( ONBOARD_STORE ).getSelectedSiteLogo();
 
-	const postSiteSettings = ( siteId: number ) => {
-		const siteSettings = {
+	if ( siteId && flowName ) {
+		const formData: ( string | File )[][] = [];
+		const settings: {
+			blogname: string;
+			blogdescription: string;
+			launchpad_screen?: string;
+			site_intent?: string;
+		} = {
 			blogname: siteTitle,
 			blogdescription: siteDescription,
 		};
 
-		return saveSiteSettings( siteId, siteSettings );
-	};
-
-	const postSiteLogo = () => {
-		if ( ! siteLogo || ! siteId ) {
-			return Promise.resolve();
-		}
-		return uploadAndSetSiteLogo(
-			siteId,
-			new File( [ base64ImageToBlob( siteLogo ) ], 'site-logo.png' )
-		);
-	};
-
-	const setIntent = ( siteId: number, flow: string ) => {
-		if ( isNewsletterOrLinkInBioFlow( flow ) ) {
+		if ( isNewsletterOrLinkInBioFlow( flowName ) ) {
 			// link-in-bio and link-in-bio-tld are considered the same intent.
-			if ( isLinkInBioFlow( flow ) ) {
-				return setIntentOnSite( siteId.toString(), LINK_IN_BIO_FLOW );
+			if ( isLinkInBioFlow( flowName ) ) {
+				settings.site_intent = LINK_IN_BIO_FLOW;
+				if ( selectedPatternContent ) {
+					const pattern: {
+						content: string | undefined;
+						template?: string | undefined;
+					} = {
+						content: selectedPatternContent,
+					};
+					pattern.template = 'blank';
```

Comment:
Hi Emanuele, Thank you for woking on this one.

Once template 'blank' is hardcoded for now, we can create the object already with that value instead of assigning it later.
What do you think? let me know if you want me to do this change or if you would like to handle yourself.

It could be like

```js
const pattern: {
     content: string | undefined;
     template?: string | undefined;
} = {
      content: selectedPatternContent,
      template:'blank'
};
```

Reply:
Thank you Renan!

I updated the code to reflect the changes we made to the new API. I also removed the type declaration, as it is no useful in our case.